### PR TITLE
chkbit 4.2.1 (new formula)

### PR DESCRIPTION
**Pull request by laktak**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/chkbit.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    # test operation
+    mkdir testpath/"foo" do
+      (testpath/"foo/one.txt").write <<~EOS
+        testing
+        testing
+        testing
+      EOS
+      system bin/"chkbit", "-u", testpath/"foo/"
+      assert_predicate testpath/"foo/.chkbit", :exist?
+    end
```
Comment (chenrui333) [inline Formula/c/chkbit.rb:37]: you dont need `foo` folder at all?

File: Formula/c/chkbit.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    url "https://files.pythonhosted.org/packages/b0/8d/43eafa8a785547c33b611068ffd6d914f5c5f96637d5b453abc556f095a0/blake3-0.4.1.tar.gz"
+    sha256 "0625c8679203d5a1d30f859696a3fd75b2f50587984690adab839ef112f4c043"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_match(/usage: chkbit \[-h\].*/, shell_output("#{bin}/chkbit"))
```
Comment (chenrui333) [inline Formula/c/chkbit.rb:26]: can we check version in here?

File: Formula/c/chkbit.rb (hunk 1 above)
Comment (laktak) [reply Formula/c/chkbit.rb]: I've removed the folder.

File: Formula/c/chkbit.rb (hunk 2 above)
Comment (laktak) [reply Formula/c/chkbit.rb]: Yes, I've added a version check.

**Comment by laktak** [comment]
@chenrui333  I think I've resolved all requested changes. Please let me know if you run into anything else.

File: Formula/c/chkbit.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/chkbit --version").chomp
+
+    # test operation
+    mkdir testpath do
```
Comment (chenrui333) [inline Formula/c/chkbit.rb:29]: testpath is always there, you dont need to mk one

File: Formula/c/chkbit.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
+class Chkbit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Check your files for data corruption"
+  homepage "https://github.com/laktak/chkbit-py"
+  url "https://github.com/laktak/chkbit-py/archive/refs/tags/v4.2.1.tar.gz"
```
Comment (chenrui333) [inline Formula/c/chkbit.rb:6]: looks like this is not on pypi? (can you switch the source to pypi one, we'd prefer pypi over github tarball)

image

File: Formula/c/chkbit.rb [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    # test operation
+    mkdir testpath do
+      (testpath/"one.txt").write <<~EOS
+        testing
+        testing
+        testing
+      EOS
+      system bin/"chkbit", "-u", testpath
+      assert_predicate testpath/".chkbit", :exist?
```
Comment (chenrui333) [inline Formula/c/chkbit.rb:36]:
```suggestion
    (testpath/"one.txt").write <<~EOS
      testing
      testing
      testing
    EOS

    system bin/"chkbit", "-u", testpath
    assert_predicate testpath/".chkbit", :exist?
```

**Comment by chenrui333** [comment]
can you also add `chkbit` into autobump.txt as well?

File: Formula/c/chkbit.rb (hunk 4 above)
Comment (laktak) [reply Formula/c/chkbit.rb]: No problem, I've release it on pypi and switched the source.

**Comment by laktak** [comment]
@chenrui333 
> can you also add `chkbit` into autobump.txt as well?

ah, that is how this works :)

I've accepted your change, switched to pipy and added it to autobump.txt.

**Review by branchv — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@laktak, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**